demos/using-dev-build/README.md

```diff
@@ -1,16 +1,63 @@
+- `whoami`: A simple canister that checks that calls are authenticated, and that returns the "principal of the caller". The implementation is terribly simple:
+  ```motoko
+  actor {
+      public query ({caller}) func whoami() : async Principal {
+          return caller;
+      };
+  };
+  ```
+  On the IC, principals are basically the "identity" of someone performing a request or "call" (hence "caller"), and every call must be authenticated — i.e., must have a valid principal. When using Internet Identity you are using [self-authenticating principals](https://smartcontracts.org/docs/interface-spec/index.html#principal), which is a very fancy way of saying that you have a private key on your laptop (hidden behind TouchID, Windows Hello, etc) that your browser uses to sign and prove that you are indeed the person issuing the calls to the IC. If the IC actually lets the call (request) through to the `whoami` canister, it means that everything checked out, and the `whoami` canister just responds with the information the IC adds to requests, namely your identity (principal).
+
+Let's now use those canisters. Don't care about details? Skip to the [helpers](#helpers).
+
+In order to talk to those canisters (for instance to view the webapp in your browser) you need to figure the ID of each canister and then use an URL of the form `https://localhost:8000/?canisterId=<canister ID>` (where `8000` is the port used by `dfx` to proxy calls to the replica; that port is usually specified in the `dfx.json`). You can find the canister IDs in the output of the `dfx command`, or by checking `dfx`'s "internal" (read: non-documented) state:
+
 ```
+~/internet-identity/demos/using-dev-build$ cat .dfx/local/canister_ids.json {
+  "__Candid_UI": {
+    "local": "r7inp-6aaaa-aaaaa-aaabq-cai"
+  },
+  "internet_identity": {
+    "local": "rwlgt-iiaaa-aaaaa-aaaaa-cai"
+  },
+  "webapp": {
+    "local": "rrkah-fqaaa-aaaaa-aaaaq-cai"
+  },
+  "whoami": {
+    "local": "ryjl3-tyaaa-aaaaa-aaaba-cai"
+  }
+```
+
+You might get different canister IDs (and that's totally fine). If the `webapp` canister ID is `rrkah-fqaaa-aaaaa-aaaaq-cai`, you should be able to point your browser to [`http://localhost:8000/?canisterId=rrkah-fqaaa-aaaaa-aaaaq-cai`](http://localhost:8000/?canisterId=rrkah-fqaaa-aaaaa-aaaaq-cai) to see the webapp. Hurray!
+
+![](./webapp.png)
+
+_If you actually use the webapp, make sure that the "Internet Identity URL" field points to `http://localhost:8000/?canisterId=<canister ID of the internet_identity canister>`._
+
+## Helpers
+
+Figuring the canister IDs, and using the `canisterId=...` query parameter is all a bit cumbersome. Here are some commands you might like:
+
+- `npm run start`: Build the app and serve it on `localhost:8080` with hot reload on code changes, ideal for hacking on the webapp.
```

Comment:
I got an error when running it:

```
➜  using-dev-build git:(nm-clean-up-testnet-app) ✗ npm run start

> e2e-tests@1.0.0 start
> webpack serve

[webpack-cli] Unable to load '@webpack-cli/serve' command
[webpack-cli] TypeError: options.forEach is not a function
```

I had to change `"webpack-cli": "4.9.1"`

It might be also useful to say which node version to use. Or add it in the package.json and `.npmrc`

First answer in [this stackoverflow question](https://stackoverflow.com/questions/28409883/how-to-specify-enforce-a-specific-node-js-version-to-use-in-package-json)

Reply:
Btw, I got a window error the first time I run it regarding size:

![Screenshot 2022-04-07 at 14 43 00](https://user-images.githubusercontent.com/98735375/162201920-8d0ed887-7451-45e3-ac30-08b73950a820.png)

It appears every time I refresh. I didn't try to fix it.

Reply:
The warnings were also in the console:

```
WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets:
  index.js (442 KiB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  index (442 KiB)
      index.js


WARNING in webpack performance recommendations:
You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
For more info visit https://webpack.js.org/guides/code-splitting/

webpack 5.45.1 compiled with 3 warnings in 2217 ms
```

Reply:
Thanks! Updated the dependency and starting `webpack serve` with `--no-client-overlay`.